Lib/unittest/test/testmock/testpatch.py

```diff
@@ -664,6 +664,21 @@ def test():
         test()
 
 
+    def test_patch_dict_decorator_resolution(self):
+        # bpo-35512: Ensure that patch with a string target resolves to
+        # the new dictionary during function call
+        @patch.dict('unittest.test.testmock.support.target', {'bar': 'BAR'})
+        def test():
+            self.assertEqual(support.target['foo'], 'BAZ')
+            self.assertEqual(support.target['bar'], 'BAR')
+
+        support.target = {'foo': 'BAZ'}
+        test()
+
+        self.assertEqual(support.target['foo'], 'BAZ')
+        self.assertNotIn('bar', support.target)
```

Comment:
Why not just:
`self.assertEqual(support.target, {'foo': 'BAZ'})`
?
(I guess the same goes for the assertions inside `test` too?)